implementations/java-mjs-validator/Harness.java

```diff
@@ -0,0 +1,213 @@
+					}
+					
+					if(cas.containsKey("registry")){
+						registry = getStringFromJson(cas.get("registry"));
+						LOGGER.log(Level.INFO, "registry found : " + registry);
+					}
+					JSONArray tests = (JSONArray) cas.get("tests");
+					
+					JSONArray resultArray = new JSONArray();
+					Iterator testIterator = tests.iterator(); 
+					
+					while (testIterator.hasNext()) { 
+						test = (JSONObject) testIterator.next();
+						Object obj = test.get("instance");
+						String instance = getStringFromJson(obj);
+						MainClass$ m = MainClass$.MODULE$;
+						Tuple2<Object, Json> results = m.validateInstance(schema, instance);
+						boolean value = (boolean) results._1();
+						
+						JSONObject result = new JSONObject();
+						result.put("valid", value);
+						resultArray.add(result);
+					}
+					
+					JSONObject output = new JSONObject();
+					output.put("seq", seq);
+					output.put("results", resultArray);
+					return output.toJSONString();
+				}
+				catch (OutOfMemoryError | StackOverflowError e) {
+					LOGGER.log(Level.SEVERE, "Exception occur in run command : " + cas.toJSONString() + " " + test.toJSONString(), e);
+					error = errorMsg("Error: Input is too large. Due to hardware restrictions of this application, we can only process small inputs.", seq);
+					return error;
+				}
+				catch(Exception e){
+					LOGGER.log(Level.SEVERE, "Exception occur in run command : " + cas.toJSONString() + " " + test.toJSONString(), e);
+					String msg = getDetailedMessage(e, test, schema);
+					error = errorMsg(msg, seq); //e.getMessage()
+					return error;
+				}
+			  case "stop":
+				if(started != true){
+					throw new RuntimeException("Bowtie hasn't started!");
+				}
+				System.exit(0);
+			}
+		}catch(Exception e){
+			LOGGER.log(Level.SEVERE, "Exception occur in operate : ", e);
+			error = errorMsg(e.getMessage(), -1);
+			return error;
+		}
+		error = errorMsg("Send correct command", -1);
+		return error;
+	}
+	
+	public static String errorMsg(String message, long seq){
+		JSONObject traceBack = new JSONObject(); 
+		traceBack.put("traceBack", message);
+		JSONObject error = new JSONObject(); 
+		error.put("errored", true);
+		error.put("seq", seq);
+		error.put("context",traceBack);
+		return error.toJSONString();
+	}
+	
+	public static String skipMsg(String message, long seq){
+		JSONObject traceBack = new JSONObject();
+		JSONObject error = new JSONObject();
+		error.put("skipped", true);
+		error.put("seq", seq);
+		error.put("message",message);
+		return error.toJSONString();
+	}
+	
+	public static String getDetailedMessage(Exception e, JSONObject test, String schema){
+		StringWriter sw = new StringWriter();
+		e.printStackTrace(new PrintWriter(sw));
+		Object obj = test.get("instance");
+		String instance = getStringFromJson(obj);
+		return sw.toString() + " " + test.toJSONString() + " " + instance + " " + schema;
+	}
+	
+	public static String getStringFromJson(Object obj){
```

Comment:
Isn't this function essentially reimplementing a JSON encoder? Surely that exists already no? You can't simply dump the given `Object` to JSON using an existing library being used here?

Reply:
I tried using multiple encoders but couldn't achieve the exact values required by our implementation. So I wrote a custom function to achieve this.

Reply:
That would sound like a bug in the implementation would it not? Are you not using a JSON *de*coder to deserialize what you get?

The point of the harness isn't really to pepper over bugs, if that's what's happening. I'd be a bit unwilling to maintain code like this inside the harness -- it's meant to represent what a normal user should do when using your implementation. You want a normal user to manually encode values to strings?